### ISSUE TITLE
DataViews Field rendering: Context should be not be assumed by fields when rendering

### ISSUE THREAD

**Issue by andrewhayward** [WordPress/gutenberg#56320]
Related: #55083

## What problem does this address?

Field rendering is context-unaware, and the same output is generated regardless of which layout is being used. This means that we should ensure rendered content is appropriate in any view.

For example, while `<h3>{ title }</h3>` is appropriate in a grid, it should be `<th scope="row">{ title }</th>` in a list.

<p align="center">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/37b7ca3a-d414-4bfd-b436-ec6a014f56d2" alt="The Data Views grid layout, with an object title circled in red" width="45%">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/f9542e3b-bba3-4208-a47b-7e7d672e4848" alt="The Data Views list layout, with an object title circled in red" width="45%">
</p>

## What is your proposed solution?

Layouts should render their own context (e.g. `<h3>`, `<th>`) and fields should not be opinionated on this. Currently, for example, [the list view](https://github.com/WordPress/gutenberg/blob/447a6de8eb79f307921a22fb8814cc01affded46/packages/edit-site/src/components/dataviews/view-list.js) renders its own `<td>`s, which wrap every field:
https://github.com/WordPress/gutenberg/blob/447a6de8eb79f307921a22fb8814cc01affded46/packages/edit-site/src/components/dataviews/view-list.js#L489-L507

The `DataViews` component could gain a `primaryField` prop, or the `fields` API could be adjusted to include an `isPrimary` flag, either of which layouts could use to render context appropriately...

```js
<DataViews
  ...
  fields={ fields }
  primaryField="title"
  ...
/>
```
```js
const fields = [
  ...
  {
	id: 'title',
	...
	isPrimary: true,
	...
  },
  ...
];
```

The alternative is to make sure that `field.render` checks the view type that is passed in, but that adds complexity that is probably not worth it. Every field would have to know to render a `<th>` or `<td>` in some instances, an `<h3>` in others, and as the list of available layouts changes, they would have to stay on top of that.

**Comment by oandregal** [comment]
I'm exploring a potential path at https://github.com/WordPress/gutenberg/pull/56942

**Comment by youknowriad** [comment]
titleField (primary) has been added, so this seems solved.
